lib/Conversion/TorchToTMTensor/TorchToTMTensor.cpp

```diff
@@ -52,6 +56,133 @@ using namespace mlir::torch::TMTensor;
 // that these patterns become mostly mechanical associations of
 // "aten.foo -> linalg.foo".
 
+static Attribute getNumericLimit(PatternRewriter& rewriter, Type elementType, bool getMin = true) {
+    auto bitWidth = elementType.getIntOrFloatBitWidth();
+    if (llvm::isa<mlir::IntegerType>(elementType)) {
+        if (getMin) {
+            return rewriter.getIntegerAttr(elementType, APInt::getSignedMinValue(bitWidth));
+        } else {
+            return rewriter.getIntegerAttr(elementType, APInt::getSignedMaxValue(bitWidth));
+        }
+    } else if (mlir::FloatType floatType = llvm::dyn_cast<mlir::FloatType>(elementType)) {
+        return rewriter.getFloatAttr(elementType, APFloat::getLargest(floatType.getFloatSemantics(), getMin));
+    } else {
+      llvm_unreachable("Only float/integer types are supported!");
+    }
+}
+
+// This function will reformat the `index` and `src` from torch operations 
+// like `torch.scatter` or `torch.scatter_reduce` to match the expected 
+// input for the TMScatterOp.
+static std::pair<Value,Value> formatTMTensorScatterOpInputs(
+        PatternRewriter& rewriter, Value self, Value indices, Value src, int64_t dim) {
```

Comment:
I don't think `self` is needed in this function

Reply:
Refactored to remove self from function parameters